### PR TITLE
[BEAM-12356] Close DatasetService leak as local variables

### PR DESCRIPTION
**Pull request by baeminbo**
I found some DatasetService local variables that are not closed.  This can cause the harmless error logs in BEAM-12356.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by baeminbo** [comment]
R: @chamikaramj

**Comment by baeminbo** [comment]
Run Java PreCommit

**Comment by baeminbo** [comment]
@reuvenlax I found some code locations that can cause the gPRC logs in BEAM-12356.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryQueryHelper.java [hunk 1]
```diff
@@ -96,105 +96,112 @@ public static TableReference executeQuery(
...
-    LOG.info("Query job {} completed", queryJobId);
-    return queryResultTable;
+      LOG.info("Query job {} completed", queryJobId);
+      return queryResultTable;
+
+    } catch (RuntimeException | IOException | InterruptedException e) {
+      throw e;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
```
Comment (reuvenlax) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryQueryHelper.java:204]: Are the new catch blocks needed?

**Comment by reuvenlax** [comment]
Thanks! Only one small question

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryQueryHelper.java (hunk 1 above)
Comment (baeminbo) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryQueryHelper.java:204]: The method throws `InterruptedException` and `IOException` only.  I refer to [BigQueryHelpers.getNumRows](https://github.com/apache/beam/blob/v2.38.0-RC1/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryHelpers.java#L554-L556).

Comment (reuvenlax) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryQueryHelper.java:204]: But the catch blocks were not there before, right?

Comment (baeminbo) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryQueryHelper.java:204]: `DatasetService.close()` throws `Exception` (not `IOException`) and the method `executeQuery` has only `IOException` and `InterruptedException` as throws in the method signature. Without the new catch blocks, `try (DatasetService tableService = bqServices.getDatasetService(options)) { ... }` causes compile error.

**Comment by reuvenlax** [comment]
LGTM

**Comment by reuvenlax** [comment]
LGTM

**Pull request merged**